src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -93,7 +93,9 @@ void Initialize()
 
 			_items.VectorChanged += (s, e) =>
 			{
+				_inProgressVectorChange = e;
```

Comment:
wrap this in try/finally, except on WASM (see eg https://github.com/unoplatform/uno/blob/master/src/Uno.UI/DataBinding/BindingExpression.cs#L183-L185 )